xarray/plot/plot.py

```diff
@@ -267,6 +280,8 @@ def line(darray, *args, **kwargs):
 
     _ensure_plottable(xplt)
 
+    xplt.values, xlabel = _interval_to_mid_points(xplt.values, xlabel)
```

Comment:
Rather than mutating `xplt.values`, please assign a new variable. Otherwise this could mutate an inline argument to this function.

Reply:
OK, changed it here and in lines 628f. Note that this changes the type of xplt passed to ax.plot from DataArray to np.array, but I think this shouldn't matter.